scripts/firm.py

```diff
@@ -1,18 +1,27 @@
 # firm functions
+def get_Y (L, K, alpha, A, g_y, t):
+    '''
+    Updated based on eqn 8.14 
+    '''
+    import math
+    
+    Y = A * (K ** alpha) * (math.exp(g_y * t) * L) ** (1 - alpha)
 
-def get_r(L, K, alpha, A, delta):
+    return Y
+    
+def get_r(Y, K, alpha, delta):
     '''
-    The interest rate implied by the firm FOC for the choice of capital
+    Updated based on eqn 8.16 
     '''
-    r = alpha * A * (L / K) ** (1 - alpha) - delta
+    r = alpha * (Y / K) - delta
 
     return r
 
 
-def get_w(r, alpha, A, delta):
+def get_w(Y, alpha, L):
```

Comment:
I think you are going to want to keep a function in here that takes in `r` and returns `w`.  It's ok to deviate from the text.  Think of what we did in class - we guessed `r` then needed `w` before we could solve the household problem.  And we didn't have `K`, `L`, or `Y` at that point in the solution.